### PR TITLE
Fix/login username registration

### LINKED ISSUE #2205: Auth: make username required an pass it into onboarding
## Goal

- make username required in /signup for google sign in
- check for premium and check if username is in use
- disable google sign up button if username empty
- when clicked, mark username as required
- pass username into localstorage or cookie
- when signing in with google, read from cookie or localstorage

## Status Quo

if you sign in with google, it generates an ugly username: tom-holland-owawd with a random suffix, even if you enter "tom" before clicking sign in with google

image

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

When getting-started page receives username in localStorage params, we validate if it's available and no premium then we perform a save and delete it from localStorage items.
This is required now as google/saml signup cannot update username on nexthauth path.

Fixes #2205

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [X] Add localStorage username as item, perform a full page reload on getting-started, look for user.username change in DB.

### CODE REVIEW

File: packages/lib/username.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import { NextApiRequest, NextApiResponse } from "next";
```
Comment (alannnc) [inline packages/lib/username.ts:1]: Relocated lib from website/lib, to package/lib

**Review by zomars — CHANGES_REQUESTED** [review]
See my comments on Twist

File: apps/web/pages/getting-started.tsx [hunk 2]
```diff
@@ -259,6 +253,44 @@ export default function Onboarding(props: inferSSRProps<typeof getServerSideProp
     token: string;
   }>({ resolver: zodResolver(schema), mode: "onSubmit" });
 
+  const fetchUsername = async (username: string) => {
+    const response = await fetch(`${process.env.NEXT_PUBLIC_APP_URL}/api/username`, {
```
Comment (alannnc) [inline apps/web/pages/getting-started.tsx:257]: @zomars Just checking out if this value it's the right for the job?

Comment (zomars) [reply apps/web/pages/getting-started.tsx:257]: Yup

**Review by zomars — APPROVED** [review]

**Pull request merged**